test/integration/IntegrationBase.t.sol

```diff
@@ -0,0 +1,920 @@
+    }
+
+    /// @dev Check that specific weights were added to the operator and total stakes for each quorum
+    function assert_Snap_AddedWeightToStakes(
+        User user, 
+        bytes memory quorums,
+        uint96[] memory addedWeights,
+        string memory err
+    ) internal {
+        uint96[] memory curOperatorStakes = _getStakes(user, quorums);
+        uint96[] memory prevOperatorStakes = _getPrevStakes(user, quorums);
+
+        uint96[] memory curTotalStakes = _getTotalStakes(quorums);
+        uint96[] memory prevTotalStakes = _getPrevTotalStakes(quorums);
+
+        for (uint i = 0; i < quorums.length; i++) {
+            assertEq(curOperatorStakes[i], prevOperatorStakes[i] + addedWeights[i], err);
+            assertEq(curTotalStakes[i], prevTotalStakes[i] + addedWeights[i], err);
+        }
+    }
+
+    /// @dev Check that the operator's stake weight was added to the operator and total
+    /// stakes for each quorum
+    function assert_Snap_Added_OperatorWeight(
+        User user, 
+        bytes memory quorums,
+        string memory err
+    ) internal {
+        uint96[] memory addedWeights = _getWeights(user, quorums);
+
+        assert_Snap_AddedWeightToStakes(user, quorums, addedWeights, err);
+    }
+
+    /// @dev For each churned quorum, checks that the incomingOperator's weight was added
+    /// and the corresponding churned operator's weight was removed
+    function assert_Snap_Churned_OperatorWeight(
+        User incomingOperator,
+        User[] memory churnedOperators,
+        bytes memory churnedQuorums,
+        string memory err
+    ) internal {
+        // Sanity check input lengths
+        assertEq(churnedOperators.length, churnedQuorums.length, "assert_Snap_Churned_OperatorWeight: input length mismatch");
+
+        // Get weights added and removed for each quorum
+        uint96[] memory addedWeights = _getWeights(incomingOperator, churnedQuorums);
+        uint96[] memory removedWeights = new uint96[](churnedOperators.length);
+        for (uint i = 0; i < churnedOperators.length; i++) {
+            removedWeights[i] = _getWeight(uint8(churnedQuorums[i]), churnedOperators[i]);
+        }
+
+        uint96[] memory curIncomingOpStakes = _getStakes(incomingOperator, churnedQuorums);
+        uint96[] memory prevIncomingOpStakes = _getPrevStakes(incomingOperator, churnedQuorums);
+
+        uint96[] memory curTotalStakes = _getTotalStakes(churnedQuorums);
+        uint96[] memory prevTotalStakes = _getPrevTotalStakes(churnedQuorums);
+
+        // For each quorum, check that the incoming operator's individual stake was increased by addedWeights
+        // and that the total stake is plus addedWeights and minus removedWeights
+        for (uint i = 0; i < churnedQuorums.length; i++) {
+            assertEq(curIncomingOpStakes[i], prevIncomingOpStakes[i] + addedWeights[i], err);
+            assertEq(curTotalStakes[i], prevTotalStakes[i] + addedWeights[i] - removedWeights[i], err);
+        }
```

Comment:
I guess you could add a check that the churned operator's stakes are now zero in the quorums?
perhaps overkill though, as this is coming quite close already 😄

Reply:
To check the snapshot version of that, this method would need to do one additional timewarp lookup per churned operator, which really slows down the tests.

I do a non-timewarp version of that in `check_Churned_State`, instead, which is where this method is called

Reply:
makes sense! yeah I would just leave as-is then.